indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java

```diff
@@ -603,6 +647,28 @@ protected void innerReduce(Context context, SortableBytes keyBytes, Iterable<Dim
     }
   }
 
+  static DimensionRangeShardSpec createShardSpec(
+      boolean isSingleDim,
```

Comment:
The number of `dimensions` should be enough to determine whether it is single dim or not. I don't think we need this argument, `isSingleDim`.

Reply:
In some cases DimensionRangePartitionsSpec can set with only one partitionDimension.  I just keep the DimensionRangePartitionsSpec what the user sets.

Reply:
In batch ingestion, if a user chooses `range` partitioning and specifies only one dimension, that results in `SingleDimensionShardSpec`s. But we can make this change later.